core/src/main/java/org/apache/iceberg/stats/BlobMetadata.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.stats;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import java.util.Set;
+import javax.annotation.Nullable;
+import org.apache.iceberg.relocated.com.google.common.base.Preconditions;
+import org.apache.iceberg.relocated.com.google.common.collect.ImmutableSet;
+
+public class BlobMetadata {
+  private final String type;
+  private final Set<Integer> columns;
+  private final long offset;
+  private final long length;
+  private final String compressionCodec;
+
+  @JsonCreator
+  public BlobMetadata(
+      @JsonProperty("type") String type,
```

Comment:
What is the benefit of the multiple blobs in a file if we don't have footer information saying why we may or may not need to check a specific blob? I guess currently we can prune them based on "columns" but I feel like maybe we would benefit from other information? Like if a particular file had sketches that applied to different row groups in a file so the correct sketch to use would be based on min and max column values or something ? Just thinking about possible future directions.

Reply:
The footer contains a type of every blob. In particular, reading application will read only the blobs of types it can utilize for current operation. Same/similar information will be added to table metadata, so that reading application doesn't need to read stats file footer if no blobs are useful to it.